### PR TITLE
Core 421

### PR DESCRIPTION
**Pull request by rtoepfer**

### CODE REVIEW

File: SLAM_example.cpp [hunk 1]
```diff
@@ -88,15 +88,15 @@ int main(int argc, char *argv[]) {
     auto status = core->startScanning();
     if(status != Status::SUCCESS) {
       std::cerr << "ERROR starting scanner" << std::endl;
-      return (int) status;
+      abort();
```
Comment (hcwiley) [inline SLAM_example.cpp]: just curios, why abort instead of return status?

File: basic_usage_mesh_generate.cpp [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    }
+
+    // continue to scan until user cancels
+    std::cout << "Press 'q' to stop scanning: ";
+    int ch;
+    do {
+      std::cin >> ch;
+      std::cout << ch;
+    }
+    while(ch != 'q');
```
Comment (hcwiley) [inline basic_usage_mesh_generate.cpp]: this isn't working as desire on my iMac:

```
blu-mac:build hcwiley$ ./basic_usage_mesh_generate
debug: Add srcs
debug: CL_DEVICE_MAX_MEM_ALLOC_SIZE: 268435456
debug: max_resolution: 322.000000
debug: determined ABSOLUTE_MAX_RESOLUTION: 256
info: cat: pub, chan: color
info: cat: pub, chan: in
info: cat: pub, chan: meta
info: cat: pub, chan: vol
debug: _c354106fe9a236c9c6117cecf932ad69da100e72f9bddb40::startScanning()....
debug: cleared preview_slam_threads
warn: _20e663132887a33f779dcba1c87d57 "dbg":"rng" was not found. (Exception std::out_of_range, what: map::at:  key not found)
warn: _20e663132887a33f779dcba1c87d57 "dbg":"nrm" was not found. (Exception std::out_of_range, what: map::at:  key not found)
warn: _20e663132887a33f779dcba1c87d57 "dbg":"ray" was not found. (Exception std::out_of_range, what: map::at:  key not found)
warn: _20e663132887a33f779dcba1c87d57 "dbg":"trk" was not found. (Exception std::out_of_range, what: map::at:  key not found)
warn: _20e663132887a33f779dcba1c87d57 "dbg":"pose_from_SLAM" was not found. (Exception std::out_of_range, what: map::at:  key not found)
debug: attached kinfu_slam_threads
Press 'q' to stop scanning: q
00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000.....................
```
Just keeps printing `0` until i Ctrl+C it. This happens both with pico plugged in and not.

File: SLAM_example.cpp (hunk 1 above)
Comment (rtoepfer) [inline SLAM_example.cpp]: I believe the return just exits the thread not the process.

Comment (rtoepfer) [inline SLAM_example.cpp]: http://www.cplusplus.com/reference/cstdlib/exit/ talks about not using exit and using abort for multi-thread apps.

**Comment by rtoepfer** [comment]
@hcwiley Changed the wait for 'q' section to not output the char to terminal.  Fixes the bug.

File: SLAM_example.cpp (hunk 1 above)
Comment (hcwiley) [inline SLAM_example.cpp]: 👍

**Pull request merged**